packages/react-accordion/src/components/AccordionHeader/useAccordionHeader.ts

```diff
@@ -20,36 +21,48 @@ export const accordionHeaderShorthandProps: Array<keyof AccordionHeaderSlots> =
  * @param props - AccordionHeader properties
  * @param ref - reference to root HTMLElement of AccordionHeader
  */
-export const useAccordionHeader = (props: AccordionHeaderProps, ref: React.Ref<HTMLElement>): AccordionHeaderState => {
+export const useAccordionHeader = (
+  {
+    id,
+    icon,
+    button,
+    children,
+    expandIcon,
+    inline = false,
+    size = 'medium',
+    expandIconPosition = 'start',
+    ...props
+  }: AccordionHeaderProps,
```

Comment:
```suggestion
  props: AccordionHeaderProps,
```

Let's move this destruction into function body to make it more readable.